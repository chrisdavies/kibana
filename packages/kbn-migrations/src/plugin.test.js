const Plugin = require('./plugin');

describe('Plugin.sanitize', () => {
  test('ensures that migrations are not undefined', () => {
    const plugins = [
      { id: 'a', migrations: [{ id: 'shazm', seed() {} }] },
      { id: 'b' },
      { id: 'c', mappings: { hello: { type: 'keyword ' } } },
    ];
    expect(Plugin.sanitize({ plugins })).toMatchSnapshot();
  });

  test('requires plugins to be specified', () => {
    expect(() => Plugin.sanitize({})).toThrowError();
  });

  test('requires plugins to have an id', () => {
    const plugins = [{ seed() { } }];
    expect(() => Plugin.sanitize({ plugins })).toThrowError(/"id" is required/);
  });
});
