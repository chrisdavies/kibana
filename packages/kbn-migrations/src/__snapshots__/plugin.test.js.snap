// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugin.sanitize ensures that migrations are not undefined 1`] = `
Array [
  Object {
    "id": "a",
    "mappingsChecksum": "",
    "migrations": Array [
      Object {
        "id": "shazm",
        "seed": [Function],
      },
    ],
    "migrationsChecksum": "787f9971146420f5432960083228cb45d419670a",
  },
  Object {
    "id": "c",
    "mappings": Object {
      "hello": Object {
        "type": "keyword ",
      },
    },
    "mappingsChecksum": "a73090e20fb1de1d847a851520ab731a6946402b",
    "migrations": Array [],
    "migrationsChecksum": "989db2448f309bfdd99b513f37c84b8f5794d2b5",
  },
]
`;
